Tests/StoreKitUnitTests/PurchasesOrchestratorTests.swift

```diff
@@ -9,246 +9,55 @@
-                                                  operationDispatcher: self.operationDispatcher,
-                                                  receiptFetcher: self.receiptFetcher,
-                                                  receiptParser: self.receiptParser,
-                                                  transactionFetcher: self.mockTransactionFetcher,
-                                                  customerInfoManager: self.customerInfoManager,
-                                                  backend: self.backend,
-                                                  transactionPoster: self.transactionPoster,
-                                                  currentUserProvider: self.currentUserProvider,
-                                                  transactionsManager: self.transactionsManager,
-                                                  deviceCache: self.deviceCache,
-                                                  offeringsManager: self.mockOfferingsManager,
-                                                  manageSubscriptionsHelper: self.mockManageSubsHelper,
-                                                  beginRefundRequestHelper: self.mockBeginRefundRequestHelper,
-                                                  storeKit2TransactionListener: storeKit2TransactionListener,
-                                                  storeKit2StorefrontListener: storeKit2StorefrontListener,
-                                                  storeMessagesHelper: self.mockStoreMessagesHelper)
-        self.storeKit1Wrapper.delegate = self.orchestrator
-    }
-
-    var transactionPoster: TransactionPoster {
-        return .init(
-            productsManager: self.productsManager,
-            receiptFetcher: self.receiptFetcher,
-            transactionFetcher: self.mockTransactionFetcher,
-            backend: self.backend,
-            paymentQueueWrapper: self.paymentQueueWrapper,
-            systemInfo: self.systemInfo,
-            operationDispatcher: self.operationDispatcher
-        )
-    }
+//  Created by Mark Villacampa on 20/2/24.
 
-}
+protocol PurchasesOrchestratorTests {
+
+    // MARK: - StoreFront Changes
+
+    func testClearCachedProductsAndOfferingsAfterStorefrontChanges() async throws
+
+    // MARK: - Purchasing
+
+    func testPurchasePostsReceipt() async throws
+
+    func testPurchaseReturnsCorrectValues() async throws
+
+    func testPurchaseDoesNotPostReceiptIfPurchaseFailed() async throws
+
+    func testPurchaseWithInvalidPromotionalOfferSignatureFails() async throws
+
+    // MARK: - Paywalls
+
+    func testPurchaseWithPresentedPaywall() async throws
+
+    func testPurchaseFailureRemembersPresentedPaywall() async throws
+
+    // MARK: - AdServices and Attributes
+
+    func testPurchaseDoesNotPostAdServicesTokenIfNotEnabled() async throws
+
+    #if !os(tvOS) && !os(watchOS)
+    @available(iOS 14.3, macOS 11.1, macCatalyst 14.3, *)
```

Comment:
Swift was complaining about marking this method unavailable in tvos/watchos in a protocol, better to just conditionally compile it